File: codewars/8-kyu/calculate-bmi/solution.py
# https://www.codewars.com/kata/57a429e253ba3381850000fb

# solution 1: 889ms
def bmi(weight, height):
    bmi = (weight/height**2)
    if bmi <= 18.5:
        return "Underweight"
    elif bmi <= 25.0:
        return "Normal"
    elif bmi <= 30.0:
        return "Overweight"
    else:
        return "Obese"

# solution 2: 576ms
def bmi(weight, height):
    b = weight / height ** 2
    return ['Underweight', 'Normal', 'Overweight', 'Obese'][(b > 30) + (b > 25) + (b > 18.5)]

File: codewars/8-kyu/calculate-bmi/solution.rs
// https://www.codewars.com/kata/57a429e253ba3381850000fb

fn bmi(weight: u32, height: f32) -> &'static str {
    let ratio = weight as f32 / height.powi(2);
    match ratio {
        ratio if ratio <= 18.5 => "Underweight",
        ratio if ratio <= 25.0 => "Normal",
        ratio if ratio <= 30.5 => "Overweight",
        _ => "Obese"
    }
}